device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceClientManagerSample.java

```diff
@@ -30,38 +39,89 @@
      */
     public static void main(String[] args)
             throws URISyntaxException, IOException {
-        log.debug("Starting the sample...");
-        log.debug("Using communication protocol: {}", protocol.name());
 
-        DeviceClient client = new DeviceClient(deviceConnectionString, protocol);
-        log.debug("Successfully created an IoT Hub client.");
+        System.out.println("Starting...");
+        
+        String argDeviceConnectionString = (args.length >= 1) ? args[0] : DEVICE_CONNECTION_STRING;
+        
+        if (argDeviceConnectionString == null && (args.length < 1 || args.length >=5) )
+        {
+            System.out.format("[Error] " + CMD_ARG + "\n" + CMD_HELP, args.length);
+            return;
+        }
+
+        System.out.println("Setup parameters...");
+        System.out.format("Setup parameter: Connection String from %s\n", (args.length >= 1) ? "command line" : "environment variable (\"IOTHUB_DEVICE_CONNECTION_STRING\")");
+
+        IotHubClientProtocol argProtocol;
+        if (args.length >= 2)
+        {
+            switch (args[1].toLowerCase())
```

Comment:
Is there a good parameter parsing framework available in Java we should use? I hate to hand-roll parameter parsing.

Reply:
args4j, MIT license.

Reply:
Commons CLI is what we went with in our track 2 SDK samples. Example can be seen [here](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/helpers/SamplesArguments.java). Both Commons CLI and Picoli look good to me, though.